dll/win32/browseui/shellbrowser.cpp

```diff
@@ -813,7 +827,11 @@ HRESULT CShellBrowser::BrowseToPIDL(LPCITEMIDLIST pidl, long flags)
     if (FAILED_UNEXPECTEDLY(hResult))
         return hResult;
 
-    newFolderSettings.ViewMode = FVM_ICON;
+    if (_ILIsSpecialFolder(pidl, CSIDL_DRIVES) || _ILIsSpecialFolder(pidl, CSIDL_CONTROLS)
+        || _ILIsSpecialFolder(pidl, CSIDL_DESKTOP))
```

Comment:
I wonder if there's simpler way to check if the folder is special. 🤔

Reply:
In Wine, they compare the GUIDs to see if they are equal but getting the GUID from the PIDL was complicated as well. [Raymond Chen recommends using ```IShellFolder::CompareIDs```](https://stackoverflow.com/questions/3572220/recognize-windows-shell-special-folder-i-e-get-its-csidl-via-its-pidl-now-de#comment9256003_7624031) for this purpose so I opted to use that.

Reply:
I don't think that this is the way it should be determined what view this folder needs, it should probably be some property from the IShellFolder or one of its related IShellFolder2 or whatever

Reply:
There is an `GetViewModeAndIconSize` in the `IFolderView2` class, and I assume that this is what should be used here.
How that determines the values is something I don't know yet.

Reply:
```IFolderView2``` is NT6+ https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ifolderview2#requirements

I would love to get that implemented eventually, it just goes well beyond the scope of this PR.